### ISSUE TITLE
Simple Test example ios/android

### ISSUE THREAD

**Issue by kuhnroyal** [badoo/Reaktive#102]
I got some stuff working on iOS/android but would like to test my multiplatform code somehow.
For example how would one go about testing the ios-app example:
```
fun calculate(): Single<List<Int>> =
    observableOf(0, 1, 2, 3, 4)
        .subscribeOn(computationScheduler)
        .doOnBeforeNext { println("Should be background thread: ${!isMainThread()}") }
        .map {
            // some off-thread computations
            sleep(1)
            it
        }
        .observeOn(mainScheduler)
        .doOnBeforeNext { println("Should be main thread: ${isMainThread()}") }
        .toList()

private fun isMainThread() = NSOperationQueue.mainQueue == NSOperationQueue.currentQueue
```

I played around with test scheduler but didn't get it to works, same with android.
I always get `The Single did not succeed: com.badoo.reaktive.test.single.TestSingleObserver`

I tried something like this:
```
val scheduler = TestScheduler()
val observable = calculate()
val sub = observable.subscribeOn(scheduler).test()
scheduler.timer.advanceBy(1000)
assertTrue(sub.isSuccess)
assertNotNull(sub.value)
assertTrue(sub.value.size == 5)
```

**Comment by arkivanov** [comment]
Hi! You should override real schedulers with `trampoline`.
Just add the following code into your test class:
```
    @BeforeTest
    fun before() {
        overrideSchedulers(
            main = ::trampolineScheduler,
            computation = ::trampolineScheduler
        )
    }

    @AfterTest
    fun after() {
        overrideSchedulers()
    }
```
It will still fail for JavaScript target, will be fixed in #103.

**Comment by kuhnroyal** [comment]
This works in a common project with tests for iOS and Android, thanks!
